prow/tide/status.go

```diff
@@ -552,6 +554,23 @@ func (sc *statusController) search() []PullRequest {
 	return prs
 }
 
+// newBaseSHAGetter is a refGetter that will look up the baseSHA from GitHub if necessary
+// and if it did so, store in in the baseSHA map
+func newBaseSHAGetter(baseSHAs map[string]string, ghc githubClient, org, repo, branch string) config.RefGetter {
+	baseSHA := baseSHAs[poolKey(org, repo, branch)]
```

Comment:
Why do we want to do this first resolution at the time we create the closure and not at the time that the closure is evaluated?

Reply:
Updated